packages/devtools_app/test/service_manager_test.dart

```diff
@@ -33,6 +33,70 @@ void main() async {
         await env.tearDownEnvironment(force: true);
       });
 
+      test('verify number of vm service calls on connect', () async {
+        await env.setupEnvironment();
+        // Await a delay to ensure the service extensions have had a chance to
+        // be called. This delay may be able to be shortened if doing so does
+        // not cause bot flakiness.
+        await Future.delayed(const Duration(seconds: 10));
+        // Ensure all futures are completed before running checks.
+        await serviceManager.service.allFuturesCompleted;
+        expect(
+          serviceManager.service.vmServiceCallCount,
+          equals(32),
+          reason:
+              'Unexpected number of vm service calls upon connection. If this '
+              'is expected, please update this test to the new expected number '
+              'of calls. Here are the calls for this test run:\n'
+              '${serviceManager.service.vmServiceCalls.toString()}',
+        );
+        expect(
+          serviceManager.service.vmServiceCalls,
+          equals(
+            [
```

Comment:
interesting!
this is a good reminder we have 3X the getIsolate we probably need at startup. Fyi @elliette

Reply:
looks like these calls are part of the test environment setup - commented as such in the test.